finagle-http/src/main/scala/com/twitter/finagle/Http.scala

```diff
@@ -107,6 +107,7 @@ object Http extends Client[Request, Response] with HttpRichClient
     val stack: Stack[ServiceFactory[Request, Response]] =
       StackClient.newStack
         .replace(TraceInitializerFilter.role, new HttpClientTraceInitializer[Request, Response])
+        .replace(StackClient.Role.prepFactory, DelayedRelease.module)
```

Comment:
Oh, interesting. finagle-http's codec _does_ install DelayedReleaseService in the prepConn:
```scala

      override def prepareConnFactory(
        underlying: ServiceFactory[Request, Response],
        params: Stack.Params
      ): ServiceFactory[Request, Response] =
        // Note: This is a horrible hack to ensure that close() calls from
        // ExpiringService do not propagate until all chunks have been read
        // Waiting on CSL-915 for a proper fix.
        underlying.map { u =>
          val filters =
            new ClientContextFilter[Request, Response].andThenIf(!_streaming ->
              new PayloadSizeFilter[Request, Response](
                params[param.Stats].statsReceiver, _.content.length, _.content.length
              )
            )

          filters.andThen(new DelayedReleaseService(u))
        }
```

Perhaps we should install DelayedRelease.module in prepConn as well?